aten/src/ATen/ScalarOps.h

```diff
@@ -11,16 +12,16 @@ namespace c10 {
 // to implement this without going through Derived Types (which are not part of core).
 inline at::Tensor scalar_to_tensor(Scalar s, const Device device = at::kCPU) {
   // This is the fast track we have for CPU scalar tensors.
-  if (device == at::kCPU) {
+  if (device == at::kCPU && !s.isComplex()) {
     if (s.isFloatingPoint()) {
-      return at::native::scalar_tensor(s, at::device(at::kCPU).dtype(at::kDouble));
```

Comment:
Q. @iseeyuan How was the existing code calling into aten::native w/o including the relevant headers first? (just curious - maybe I'm missing something here).

Reply:
all native functions are declared in this generated header: https://codebrowser.bddppq.com/pytorch/pytorch/build/aten/src/ATen/NativeFunctions.h.html